packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/dataStreamSearch/workspaceSelector/workspaceSelector.tsx

```diff
@@ -34,11 +34,12 @@ export const WorkspaceSelector = ({
 
   useEffect(() => {
     OnGettingError(status === 'error');
-  }, [status, OnGettingError]);
+    if (status === 'error') setStoredWorkspace(null);
```

Comment:
A workspace doesn't work for an account might work for another. Consider the scenario:
1. account A login and set workspace A
2. account B login and go to search but not setting a workspace
3. account A login again
Now should account A see workspace A preselected?